pkg/vgmanager/devices.go

```diff
@@ -71,119 +72,132 @@ func (r *VGReconciler) addDevicesToVG(ctx context.Context, vgs []lvm.VolumeGroup
 	return nil
 }
 
-// getAvailableDevicesForVG determines the available devices that can be used to create a volume group.
-func (r *VGReconciler) getAvailableDevicesForVG(ctx context.Context, blockDevices []lsblk.BlockDevice, vgs []lvm.VolumeGroup, volumeGroup *lvmv1alpha1.LVMVolumeGroup) ([]lsblk.BlockDevice, error) {
-	// filter devices based on DeviceSelector.Paths if specified
-	availableDevices, err := r.filterMatchingDevices(ctx, blockDevices, vgs, volumeGroup)
-	if err != nil {
-		return nil, fmt.Errorf("failed to filter matching devices for volume group %s: %w", volumeGroup.GetName(), err)
-	}
+type FilteredBlockDevice struct {
+	lsblk.BlockDevice
+	FilterErrors []error
+}
 
-	return r.filterAvailableDevices(ctx, availableDevices), nil
+type FilteredBlockDevices struct {
+	Available []lsblk.BlockDevice
+	Filtered  []FilteredBlockDevice
```

Comment:
I think renaming `Filtered` to `Excluded` would make sense here as well. Both to unify naming and also to make it more obvious what the name suggests.